### ISSUE TITLE
Ownership analysis doesn't annotate structs with derives

### ISSUE THREAD

**Issue by TheDan64** [immunant/c2rust#139]
Given the command `select ann 'crate; desc(fn || field);' \; ownership_annotate ann` and the input:

```rust
#[repr(C)]
struct Foo {
    bar: *mut u32,
}

unsafe fn foo(foo: *mut Foo) {
    *(*foo).bar = 1;
}
```

You correctly get:
```rust
#[repr(C)]
struct Foo {
    #[ownership_static(WRITE)]
    bar:  *mut u32,
}

#[ownership_constraints(le(WRITE, _0))]
#[ownership_mono("", WRITE)]
unsafe fn foo(foo:  *mut Foo)  {
    *(*foo).bar = 1;
}
```

However, if you add `#[derive(Copy, Clone)]` (or any other derives) to `Foo`, then there are no resulting annotations on the struct

**Comment by rinon** [comment]
Looks like the derive macro expansion drops marks in the struct, so they aren't marked when the ownership annotation pass happens.

**Comment by rinon** [comment]
Should be fixed by 537a387d, please confirm.
